Add MODEL_BCC to enable BCC-GEOS-Chem v2.0

sunrz7 opened the pull request:
### Name and Institution (Required)

Name: Sun Ruize
Institution: Peking University

### Describe the update

This update will be used for the integration of GEOS-Chem with Beijing Climate Centre Earth System Model. We add MODEL_BCC for the convenience of coupling code. 


### Reference(s)

In preparation 

### Related Github Issue

N/A

yantosca commented on GeosCore/aerosol_mod.F90:
```diff
@@ -280,8 +282,13 @@ SUBROUTINE AEROSOL_CONC( Input_Opt,  State_Chm, State_Diag, &
+#else
+    FND = .True.
+    RC  = GC_SUCCESS
+#endif
```
@sunrz7: I think we can simplify the logic here:
```F90
#if !defined( MODEL_BCC ) && !defined( MODEL_CESM )
   CALL HCO_GC_EvalFld( Input_Opt, State_Grid, Trim(FieldName), State_Chm%OMOC, RC, FOUND=FND )
#else
    FND = .True.
    RC  = GC_SUCCESS
#endif
```
Tagging @lizziel for a second opinion.

yantosca commented on GeosCore/aerosol_mod.F90:
```diff
@@ -124,9 +124,11 @@ SUBROUTINE AEROSOL_CONC( Input_Opt,  State_Chm, State_Diag, &
 !
     USE ErrCode_Mod
     USE ERROR_MOD
+#if !defined( MODEL_BCC )
```
@sunrz7 We can collapse this into a single #if block:
```F90
#if !defined( MODEL_BCC ) && !defined( MODEL_CESM )
   ...
#endif
```

yantosca commented on GeosCore/calc_met_mod.F90:
```diff
@@ -336,12 +336,16 @@ SUBROUTINE AIRQNT( Input_Opt, State_Chm, State_Grid, State_Met, &
        ! (lower edge, delta, centroid, and spatially-weighted mean)
        !=============================================================
-
+#if !defined (MODEL_BCC) 
        ! Pressure at bottom edge of grid box [hPa]
        State_Met%PEDGE(I,J,L) = GET_PEDGE(I,J,L)
 
        ! Pressure at top edge of grid box [hPa]
        PEdge_Top = GET_PEDGE(I,J,L+1)
+#endif
+#if defined (MODEL_BCC)
+        PEdge_Top=State_Met%PEDGE(I,J,L+1)
+#endif
```
@sunrz7 We can simplify the logic here:
```F90
#ifdef MODEL_BCC
       PEdge_Top=State_Met%PEDGE(I,J,L+1)
#else
       ! Pressure at bottom edge of grid box [hPa]
       State_Met%PEDGE(I,J,L) = GET_PEDGE(I,J,L)

       ! Pressure at top edge of grid box [hPa]
       PEdge_Top = GET_PEDGE(I,J,L+1)
#endif
```

yantosca commented on GeosCore/sulfate_mod.F90:
```diff
@@ -2590,7 +2591,7 @@ SUBROUTINE CHEM_SO2( Input_Opt, State_Chm,  State_Diag, State_Grid, &
       Ptr2D => NULL()
 #endif
-
+#endif
```
@sunrz7: We can simplify the logic here:
```F90
#if !defined( MODEL_BCC ) &&  !defined( MODEL_CESM )
      CALL HCO_GC_GetDiagn( Input_Opt, State_Grid, 'SEASALT_DENS_FINE', &
                       StopIfNotFound=.FALSE., RC=RC, Ptr2D=Ptr2D )
      ENDIF
      Ptr2D => NULL()
#endif
```

lizziel reviewed (commented):
Thanks for the updates. The CPP code can be consolidated a bit. Please also add an update to the Changelog.

sunrz7 commented:
Thanks you very much! I have modified according to your suggestions. And maybe a briefly description can be: Added updates for compatibility with BCC.
If more specific description needed, please let me know!

yantosca commented:
Thanks @sunrz7.  Would you be able to edit the CHANGELOG.md yourself and then push it to your branch?  Thanks.

sunrz7 commented:
Thanks. I have edited the CHANGELOG.md and push it!

lizziel commented on CHANGELOG.md:
```diff
@@ -3,6 +3,9 @@
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/), and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+## [14.6.0] - 2025-03-31
+### Added
+- Added updates for compatibility with BCC.
```
Please writing out what BCC is since not all users will know.

yantosca replied:
@lizziel: Now fixed in [371417c](https://github.com/geoschem/geos-chem/pull/2791/commits/371417cb60f87eb154838dd5c434f750a2ae7796)

lizziel reviewed (changes requested):
Hi @sunrz7, I just have one request to update the changelog to not use BCC acronym. Not all users will know what this is so it is better to be explicit.

lizziel reviewed (approved):
Thanks @sunrz7!

sunrz7 commented:
> Hi @sunrz7, I just have one request to update the changelog to not use BCC acronym. Not all users will know what this is so it is better to be explicit.

Thanks. I think a better way is : Added updates for compatibility with Beijing Climate Centre Earth System Model. I think We should put away version 2.0 because v2.0 is the version of the coupling model (BCC-GEOS-Chem).
